SU2_CFD/src/integration/CIntegration.cpp

```diff
@@ -59,7 +59,7 @@ void CIntegration::Space_Integration(CGeometry *geometry,
       solver_container[MainSolver]->Upwind_Residual(geometry, solver_container, numerics, config, iMesh);
       break;
     case FINITE_ELEMENT:
-      solver_container[MainSolver]->Convective_Residual(geometry, solver_container, numerics[CONV_TERM], config, iMesh, iRKStep);
+      SU2_MPI::Error("Computation of FEM convective flux in non-FEM integration attempted.", CURRENT_FUNCTION);
       break;
```

Comment:
Thank you for going after dead code :+1: 
It should also be fairly safe to not cover the FINITE_ELEMENT case or just handle `default`, I doubt the execution will reach here if the wrong integration class is instantiated (multiple things would have to go wrong).
But looks good to me.

Reply:
By the way, there is dead code even in `SU2_CFD.cpp`, the `else` branch is unreachable: If it were reached, then `turbo` and `harmonic_balance` would both be `false`. As `(multizone && !turbo)` has to be `false`, `multizone` must be `false` as well. But then `((!multizone && !harmonic_balance && !turbo) || (turbo && disc_adj))` is `true` ;)

Reply:
I didn't touch this until now because I don't understand the principle behind these nested `if`-statement. Do you think the `else` branch in `SU2_CFD.cpp` should be removed?

Reply:
I'm not even going to try unfolding the logic, take it out or throw an error, if the tests pass it's fine by me.

Reply:
:spaghetti:

Reply:
We can even make the constructors of CFluidDriver protected since it is not meant for standalone use anymore (it exists as the base for the legacy drivers).

Reply:
done in commit 7825b3b

Reply:
> It should also be fairly safe to not cover the FINITE_ELEMENT case or just handle `default`, I doubt the execution will reach here if the wrong integration class is instantiated (multiple things would have to go wrong).

I agree, commit 42fab4f